contracts/MarginCalculator.sol

```diff
@@ -35,24 +35,76 @@ contract MarginCalculator {
 
+        // strike price is denominated in strike asset.
         uint256 strikePrice = otoken.strikePrice();
-        (uint256 underlyingPrice, bool isFinalized) = _getUnderlyingPrice(_otoken);
-        require(isFinalized, "MarginCalculator: Oracle price not finalized yet.");
+
+        // divide price of underlying by price of strike,
+        // to get the real price of underlying denominated in strike at expiry
+        (uint256 underlyingPrice, bool isUnderlyingFinalized) = _getAssetPrice(
+            otoken.underlyingAsset(),
+            otoken.expiryTimestamp()
+        );
+        (uint256 strikeAssetPrice, bool isStrikeFinalized) = _getAssetPrice(
+            otoken.strikeAsset(),
+            otoken.expiryTimestamp()
+        );
+
+        require(isUnderlyingFinalized, "MarginCalculator: underlying price not finalized yet.");
+        require(isStrikeFinalized, "MarginCalculator: strike price not finalized yet.");
+
+        FPI.FixedPointInt memory underlyingPriceFixedPoint = _uint256ToFPI(underlyingPrice);
+        FPI.FixedPointInt memory strikeAssetPriceFixedPoint = _uint256ToFPI(strikeAssetPrice);
+
+        FPI.FixedPointInt memory underlyingToStrikeFixedPoint = underlyingPriceFixedPoint.div(
+            strikeAssetPriceFixedPoint
+        );
+
+        uint256 underlyingToStrike = SignedConverter.intToUint(underlyingToStrikeFixedPoint.value);
 
         if (otoken.isPut()) {
-            return strikePrice > underlyingPrice ? strikePrice.sub(underlyingPrice) : 0;
+            return strikePrice > underlyingToStrike ? strikePrice.sub(underlyingToStrike) : 0;
         } else {
-            return underlyingPrice > strikePrice ? underlyingPrice.sub(strikePrice) : 0;
+            return underlyingToStrike > strikePrice ? underlyingToStrike.sub(strikePrice) : 0;
         }
     }
 
+    /**
+     * @notice Return the net worth of an expired oToken in collateral.
+     * @param _otoken otoken address
+     * @return the exchange rate that shows how much collateral unit can be take out by 1 otoken unit, scaled by 1e18
+     */
+    function getExpiredPayoutRate(address _otoken) external view returns (uint256) {
+        uint256 cashValueInStrike = _getExpiredCashValue(_otoken);
+
+        OtokenInterface otoken = OtokenInterface(_otoken);
+        address strike = otoken.strikeAsset();
+        address collateral = otoken.collateralAsset();
+
+        uint256 exchangeRate;
+
+        if (strike == collateral) {
+            exchangeRate = cashValueInStrike;
+        } else {
+            uint256 expiry = otoken.expiryTimestamp();
+            // isFinalized of strike asset is already checked in _getExpiredCashValue
+            (uint256 strikePrice, ) = _getAssetPrice(strike, expiry);
+            (uint256 collateralPrice, bool isCollateralPriceFinalized) = _getAssetPrice(collateral, expiry);
+            require(isCollateralPriceFinalized, "MarginCalculator: collateral price is not finalized");
```

Comment:
don't you also need strikePrice to be finalized?

Reply:
isFinalized of strike asset is already checked in _getExpiredCashValue